src/System.Linq/src/System/Linq/Lookup.cs

```diff
@@ -247,8 +245,7 @@ public IEnumerable<TResult> ApplyResultSelector<TResult>(Func<TKey, IEnumerable<
                 do
                 {
                     g = g._next;
-                    g.Trim();
-                    yield return resultSelector(g._key, g._elements);
+                    yield return resultSelector(g._key, g);
```

Comment:
> Also, while it is unfortunate to expose the array, there is a risk now that someone took dependency on that.

Is there really? I mean Linq is generally not that high-perf anyways so it's hard for me to imagine someone (purposefully or accidentally) casting the parameter to an array to get better perf.

> Another reason might be to separate keys from elements before passing to the selector - that is just to ensure if eventually only elements are retained, they do not hold the keys as well.

Good point. I think we can fix this and still avoid trimming by passing in a `new ArraySegment<T>(g._elements, 0, g._count)`, though.